dask/array/tests/test_ghost.py

```diff
@@ -160,3 +163,102 @@ def test_ghost():
 
     g = ghost(d, depth={0: 2, 1: 1}, boundary={0: 100})
     assert g.chunks == ((8, 8), (5, 5))
+
+
+def test_map_overlap():
+    x = da.arange(10, chunks=5)
+
+    y = x.map_overlap(lambda x: x + len(x), depth=2)
+    assert eq(y, np.arange(10) + 5 + 2 + 2)
+
+
+def test_nearest_ghost():
+    a = np.arange(144).reshape(12, 12).astype(float)
+
+    darr = da.from_array(a, chunks=(6, 6))
+    garr = ghost(darr, depth={0: 5, 1: 5},
+                 boundary={0: 'nearest', 1: 'nearest'})
+    tarr = trim_internal(garr, {0: 5, 1: 5})
+    assert_array_almost_equal(np.array(tarr), a)
+
+
+def test_0_depth():
+    expected = np.arange(100).reshape(10, 10)
+    darr = da.from_array(expected, chunks=(5, 2))
+
+    depth = {0: 0, 1: 0}
+
+    reflected = ghost(darr, depth=depth, boundary='reflect')
+    constant = ghost(darr, depth=depth, boundary=42)
+
+    result = trim_internal(reflected, depth)
+    assert_array_equal(result, expected)
+
+    result = trim_internal(constant, depth)
+    assert_array_equal(result, expected)
+
+
+@pytest.mark.xfail
+def test_0_depth_failes():
```

Comment:
Once these are working I think this will be ready for review. Also spelling error.